Sources/BrowserServicesKit/Common/Extensions/URLExtension.swift

```diff
@@ -137,6 +137,37 @@ extension URL {
         })
         return queryItem?.value
     }
+    
+    static func trimHostToETLD(host: String) -> String {
+        guard !host.isEmpty else {
+            return host
+        }
+        
+        var newHost = host
+        while newHost.contains(".") {
+            let comps = newHost.split(separator: ".").dropFirst()
+            if comps.count == 1 || comps.joined(separator: ".") == "co.uk" {
+                break
+            }
+            
+            newHost = comps.joined(separator: ".")
+        }
+        
+        return newHost
+    }
```

Comment:
Please use TLD class instead of this. :)

Reply:
I figured there had to be something for this. I'll have a look. It was on my list of things to resolve 🙂